projects/observability/src/shared/dashboard/widgets/table/table-widget-renderer.component.ts

```diff
@@ -519,6 +541,62 @@ export class TableWidgetRendererComponent
      */
   }
 
+  public onDownloadCsvSelected(): void {
+    //Todo fix this
+    this.confirmationService
+    .confirm({ content: this.downloadAsCsvTemplate })
+    .pipe(
+      filter(confirm => confirm),
+      switchMap(() => {
+        const csvDownloadFileConfig: CsvDownloadFileConfig = this.getCsvDownloadFileConfig();
+        return !isNil(csvDownloadFileConfig)
+          ? this.fileDownloadService.downloadAsCsv(csvDownloadFileConfig)
+          : throwError('No data available.')
+      })
+    )
+    .subscribe();
+  }
+
+  private getCsvDownloadFileConfig(): CsvDownloadFileConfig {
+    return {
+      fileName: `explore_results.csv`,
+      dataSource: combineLatest([
+        this.model.getData(),
+        this.columnConfigs$.pipe(map(columns => columns.filter(column => column.visible)))
+      ]).pipe(
+        switchMap(([data, columns])  => this.getDownloadableData(data, columns)),
+        map(result => result.data.map(row => {
+            const eventAttributes: Dictionary<unknown> = {};
+            Object.keys(row).forEach(key => {
+              if(typeof row[key] !== 'object') {
```

Comment:
the data contained could have an object/string/date or any kind of type, for this reason, is difficult to make it fit in the cvs.